include/ginkgo/c_api.h

```diff
@@ -0,0 +1,851 @@
+    typedef struct gko_matrix_dense_##_name##_st* gko_matrix_dense_##_name;  \
+    gko_matrix_dense_##_name ginkgo_matrix_dense_##_name##_create(           \
+        gko_executor exec, gko_dim2_st size);                                \
+    gko_matrix_dense_##_name ginkgo_matrix_dense_##_name##_create_view(      \
+        gko_executor exec, gko_dim2_st size, _ctype* values, size_t stride); \
+    void ginkgo_matrix_dense_##_name##_delete(                               \
+        gko_matrix_dense_##_name mat_st_ptr);                                \
+    void ginkgo_matrix_dense_##_name##_fill(                                 \
+        gko_matrix_dense_##_name mat_st_ptr, _ctype value);                  \
+    _ctype ginkgo_matrix_dense_##_name##_at(                                 \
+        gko_matrix_dense_##_name mat_st_ptr, size_t row, size_t col);        \
+    gko_dim2_st ginkgo_matrix_dense_##_name##_get_size(                      \
+        gko_matrix_dense_##_name mat_st_ptr);                                \
+    _ctype* ginkgo_matrix_dense_##_name##_get_values(                        \
+        gko_matrix_dense_##_name mat_st_ptr);                                \
+    const _ctype* ginkgo_matrix_dense_##_name##_get_const_values(            \
+        gko_matrix_dense_##_name mat_st_ptr);                                \
+    size_t ginkgo_matrix_dense_##_name##_get_num_stored_elements(            \
+        gko_matrix_dense_##_name mat_st_ptr);                                \
+    size_t ginkgo_matrix_dense_##_name##_get_stride(                         \
+        gko_matrix_dense_##_name mat_st_ptr);                                \
+    void ginkgo_matrix_dense_##_name##_compute_dot(                          \
+        gko_matrix_dense_##_name mat_st_ptr1,                                \
+        gko_matrix_dense_##_name mat_st_ptr2,                                \
+        gko_matrix_dense_##_name mat_st_ptr_res);                            \
+    void ginkgo_matrix_dense_##_name##_compute_norm1(                        \
+        gko_matrix_dense_##_name mat_st_ptr1,                                \
+        gko_matrix_dense_##_name mat_st_ptr2);                               \
+    void ginkgo_matrix_dense_##_name##_compute_norm2(                        \
+        gko_matrix_dense_##_name mat_st_ptr1,                                \
+        gko_matrix_dense_##_name mat_st_ptr2);                               \
+    gko_matrix_dense_##_name ginkgo_matrix_dense_##_name##_read(             \
+        const char* str_ptr, gko_executor exec);                             \
+    char* ginkgo_matrix_dense_##_name##_write_mtx(                           \
+        gko_matrix_dense_##_name mat_st_ptr);
+
+
+/**
+ * @brief A build instruction for defining gko::matrix::Csr<Tv, Ti>, simplifying
+ * its construction by removing the repetitive typing of array's name.
+ *
+ * @param _ctype_value  Type name of the element type in C
+ *
```

Comment:
```suggestion
```
We rather keep all the @param together instead of separating them by an empty line.

Reply:
Thank you! I will edit the ones that do not match the format